pkg/vault/contracts/Router.sol

```diff
@@ -777,9 +777,7 @@ contract Router is IRouter, RouterCommon, ReentrancyGuardTransient {
                 abi.encodeCall(
                     Router.queryAddLiquidityHook,
                     AddLiquidityHookParams({
-                        // We use the Router as a sender to simplify basic query functions,
-                        // but it is possible to add liquidity to any recipient.
-                        sender: address(this),
+                        sender: sender,
```

Comment:
It'd be nice to have a mutation test, checking if each query hook is called with the right sender, but maybe for another PR.